server.py

```
@@ -638,233 +767,110 @@
-            pass
+        cache_manager = get_cache_manager()
+        if cache_manager.should_cleanup():
+            # Run maintenance in background to avoid blocking tool execution
+            import asyncio
 
-        arguments = await reconstruct_thread_context(arguments)
-        logger.debug(f"[CONVERSATION_DEBUG] After thread reconstruction, arguments keys: {list(arguments.keys())}")
-        if "_remaining_tokens" in arguments:
-            logger.debug(f"[CONVERSATION_DEBUG] Remaining token budget: {arguments['_remaining_tokens']:,}")
+            asyncio.create_task(_perform_cache_maintenance())
+    except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)